__tests__/MongoBulkDataMigration.rollback.test.ts

```diff
@@ -943,6 +943,32 @@ describe('MongoBulkDataMigration', () => {
         const restoredDocuments = await collection.find().toArray();
         expect(restoredDocuments).toEqual(insertedDocuments);
       });
+
+      it('should rollback with a custom query', async () => {
+        const document = {
+          keys: [{ subKey1: 'match_me' }, { subKey1: 'do_not_match_me' }],
+        };
+        await collection.insertMany([document]);
+
+        const migration = new MongoBulkDataMigration<any>({
+          ...DM_DEFAULT_SETUP,
+          update: {
+            $set: {
+              'keys.$[element].new_key': '____MATCHED____',
```

Comment:
@qiao360 Commit text is misleading, sorry. The test name is clearer actually.
This test is all about this line (the custom rollback)
```ts
          rollback: (backup) => ({ $set: { keys: backup.keys } }),
```

This is more a test to document how to do without the automatic rollback behavior.
- if we both update and add keys because the auto-rollback has limitations as underlined by Hugo https://github.com/360Learning/mongo-bulk-data-migration/pull/22#discussion_r1840973155
- also this can be useful if a user want to rollback long after and prefers to fully restore an array
  - this can be mandatory for arrays where objects are inserted (usually in Mongo, inner-documents are pushed), or frequently updated